src/app/registration/registration.component.ts

```diff
@@ -0,0 +1,158 @@
+import { Component, OnInit } from '@angular/core';
+import { Router }            from '@angular/router';
+import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms';
+
+import { UserInterface } from '../shared/interface/userinterface.interface';
+import { User } from '../shared/models/user';
+
+import { RegistrationDirective } from './registration.directive';
+
+import { UserService } from '../shared/services';
+
+// TODO tests form submission, mock the service
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.scss'],
+  providers: [UserService]
+})
+export class RegistrationComponent implements OnInit {
+  user: FormGroup;
+  User: UserInterface; // The UserInterface is an interface used specifcally only for forms
```

Comment:
The comment left here should be placed in `userinterface.interface.ts` to prevent future confusion.